modules/sublimeBidAdapter.js

```diff
@@ -1,136 +1,217 @@
+  }
+
+  return validBidRequests.map(bid => {
+    const bidHost = bid.params.bidHost || DEFAULT_BID_HOST;
+    const protocol = bid.params.protocol || DEFAULT_PROTOCOL;
+
+    setState({
+      transactionId: bid.transactionId,
+      zoneId: bid.params.zoneId,
+      debug: bid.params.debug || false,
+    });
 
-  /**
-     * Determines whether or not the given bid request is valid.
-     *
-     * @param {BidRequest} bid The bid params to validate.
-     * @return boolean True if this is a valid bid, and false otherwise.
-     */
-  isBidRequestValid: (bid) => {
-    return !!bid.params.zoneId;
-  },
-
-  /**
-     * Make a server request from the list of BidRequests.
-     *
-     * @param {BidRequest[]} validBidRequests An array of bids
-     * @param {Object} bidderRequest - Info describing the request to the server.
-     * @return ServerRequest Info describing the request to the server.
-     */
-  buildRequests: (validBidRequests, bidderRequest) => {
-    let commonPayload = {
-      sublimeVersion: SUBLIME_VERSION,
-      // Current Prebid params
-      prebidVersion: '$prebid.version$',
-      currencyCode: config.getConfig('currency.adServerCurrency') || DEFAULT_CURRENCY,
-      timeout: config.getConfig('bidderTimeout'),
+    const bidPayload = {
+      adUnitCode: bid.adUnitCode,
+      auctionId: bid.auctionId,
+      bidder: bid.bidder,
+      bidderRequestId: bid.bidderRequestId,
+      bidRequestsCount: bid.bidRequestsCount,
+      requestId: bid.bidId,
+      sizes: bid.sizes.map(size => ({
+        w: size[0],
+        h: size[1],
+      })),
+      transactionId: bid.transactionId,
+      zoneId: bid.params.zoneId,
     };
 
-    // RefererInfo
-    if (bidderRequest && bidderRequest.refererInfo) {
-      commonPayload.referer = bidderRequest.refererInfo.referer;
-      commonPayload.numIframes = bidderRequest.refererInfo.numIframes;
+    const payload = Object.assign({}, commonPayload, bidPayload);
+
+    return {
+      method: 'POST',
+      url: protocol + '://' + bidHost + '/bid',
+      data: payload,
+      options: {
+        contentType: 'application/json',
+        withCredentials: true
+      },
     }
-    // GDPR handling
-    if (bidderRequest && bidderRequest.gdprConsent) {
-      commonPayload.gdprConsent = bidderRequest.gdprConsent.consentString;
-      commonPayload.gdpr = bidderRequest.gdprConsent.gdprApplies; // we're handling the undefined case server side
+  });
+}
+
+/**
+ * Unpack the response from the server into a list of bids.
+ *
+ * @param {*} serverResponse A successful response from the server.
+ * @param {*} bidRequest An object with bid request informations
+ * @return {Bid[]} An array of bids which were nested inside the server.
+ */
+function interpretResponse(serverResponse, bidRequest) {
+  const bidResponses = [];
+  const response = serverResponse.body;
+
+  sendEvent('dintres');
```

Comment:
pixel fires outside of the pixel listen hooks is not allowed. please remove this.